client/state/plugins/wporg/actions.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+const debug = require( 'debug' )( 'calypso:wporg-data:actions' );
+
+/**
+ * Internal dependencies
+ */
+import wporg from 'lib/wporg';
+import utils from 'lib/plugins/utils';
+import { RECEIVE_WPORG_PLUGIN_DATA, FETCH_WPORG_PLUGIN_DATA } from 'state/action-types';
+
+/**
+ *  Local variables;
+ */
+let _fetching = {};
+
+export default {
+	fetchPluginData: function( pluginSlug ) {
+		return ( dispatch ) => {
+			if ( _fetching[ pluginSlug ] ) {
+				return;
+			}
+			_fetching[ pluginSlug ] = true;
+
+			setTimeout( () => {
```

Comment:
yup... without setTimeout:

![image](https://cloud.githubusercontent.com/assets/1554855/12480075/f4239020-c03f-11e5-9456-f03f31696f19.png)

It happens in main.jsx when you return from a single plugin view. I suspect this is because we send several requests (one for each plugin) which triggers multiple re-renders once resolved and kind of messes with everything.

Thinking about it, maybe we should rethink a little about this. Maybe in this case it should be plugin-items who fetches its own plugin info, instead of doing all the fetches from its parent and receiving an already complete plugin once the data is retrieved. I think we shouldn't address this in this PR, but we should experiment a little on that direction once this is merged. Not only to avoid this setTimeout, but also to avoid all the main list rerenders when only one of the plugin-items should be redrawed